pandas/core/groupby.py

```diff
@@ -1402,6 +1408,7 @@ def head(self, n=5):
         0  1  2
         2  5  6
         """
+        self._reset_group_selection()
```

Comment:
ok that is fine. just doc the method ``_reset_group_selection()`` (and add a doc-string to ``_set_selection_from_grouper``. Maybe rename these to be more consistent as well

```
_set_group_selection()
_reset_group_selection()
``